mx_microbench.py

```diff
@@ -0,0 +1,112 @@
+#
+# ----------------------------------------------------------------------------------------------------
+#
+# Copyright (c) 2016, 2016, Oracle and/or its affiliates. All rights reserved.
+# DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+#
+# This code is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 2 only, as
+# published by the Free Software Foundation.
+#
+# This code is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+# version 2 for more details (a copy is included in the LICENSE file that
+# accompanied this code).
+#
+# You should have received a copy of the GNU General Public License version
+# 2 along with this work; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+# or visit www.oracle.com if you need additional information or have any
+# questions.
+#
+# ----------------------------------------------------------------------------------------------------
+
+import mx
+from argparse import ArgumentParser
+
+_microbench_executor = None
+
+def set_microbenchmark_executor(ex):
+    global _microbench_executor
+    assert _microbench_executor is None, 'cannot override microbenchmark executor twice'
```

Comment:
It two downstream projects, say `graal-core` and `truffle`, want to change the executor we have a problem. But the same is true for the unittest vm launcher. I guess we are fine for the moment. If this turns out to be a problem we can change it.